### ISSUE TITLE
Aeon Reading Room request getting a 404

### ISSUE THREAD

**Issue by maxkadel** [pulibrary/orangelight#3604]
### Expected behavior
When you click on "Reading Room Request" from a special collections item, it should take you to a pre-filled form in Aeon, based on the generated OpenUrl. 

### Actual behavior
When you click on "Reading Room Request" it takes you to an Aeon 404 page
![image.png](https://images.zenhubusercontent.com/5d51624a5b89150001ae0b03/ac364e02-e708-4318-83fa-a94e8d2a083c)

### Steps to replicate
1. Go to https://catalog.princeton.edu/catalog/9947127883506421
2. Click on "Reading Room Request"

### Impact of this bug
It is not clear how many objects or which objects this is affecting. People have to manually create Aeon requests, and if they're not familiar with the system, they may not know how to get to the place where they can create manual requests. 

Looking at a number of items in the `rare$ctsn` location, and not seeing similar behavior.

URL generated as of ticket writing
```
https://lib-aeon.princeton.edu/aeon/aeon.dll/OpenURL?url_ver=Z39.88-2004&url_ctx_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Actx&ctx_ver=Z39.88-2004&ctx_tim=2023-06-05T14%3A24%3A46%2B00%3A00&ctx_id=&ctx_enc=info%3Aofi%2Fenc%3AUTF-8&rft.genre=book&rft.au=Hau%CC%88y%2C+Valentin&rft.aucorp=A+Paris%3A+Imprime%CC%81+par+les+enfans-aveugles%2C+sous+la+direction+de+M.+Clousier%2C+imprimeur+du+Roi%3B+%26+se+vend%2C+a%CC%80+leur+be%CC%81ne%CC%81fice%2C+en+leur+maison+d%27education%2C+rue+Notre-Dame-des-Victoires&rft.pub=A+Paris%3A+Imprime%CC%81+par+les+enfans-aveugles%2C+sous+la+direction+de+M.+Clousier%2C+imprimeur+du+Roi%3B+%26+se+vend%2C+a%CC%80+leur+be%CC%81ne%CC%81fice%2C+en+leur+maison+d%27education%2C+rue+Notre-Dame-des-Victoires&rft.date=1786&rft.title=Essai+sur+l%27e%CC%81ducation+des+aveugles%2C+ou%2C+Expose%CC%81+de+diffe%CC%81rens+moyens%2C+ve%CC%81rifie%CC%81s+par+l%27expe%CC%81rience%2C+pour+les+mettre+en+e%CC%81tat+de+lire%2C+a%CC%80+l%27aide+du+tact%2C+d%27imprimer+des+livres+dans+lesquels+ils+puissent+prendre+des+connoissances+de+langues%2C+d%27histoire%2C+de+ge%CC%81ographie%2C+de+musique%2C+%26c.%2C+d%27exe%CC%81cuter+diffe%CC%81rens+travaux+relatifs+aux+me%CC%81tiers%2C+%26c.+%3A+de%CC%81die%CC%81+au+Roi&rft.btitle=Essai+sur+l%27e%CC%81ducation+des+aveugles%2C+ou%2C+Expose%CC%81+de+diffe%CC%81rens+moyens%2C+ve%CC%81rifie%CC%81s+par+l%27expe%CC%81rience%2C+pour+les+mettre+en+e%CC%81tat+de+lire%2C+a%CC%80+l%27aide+du+tact%2C+d%27imprimer+des+livres+dans+lesquels+ils+puissent+prendre+des+connoissances+de+langues%2C+d%27histoire%2C+de+ge%CC%81ographie%2C+de+musique%2C+%26c.%2C+d%27exe%CC%81cuter+diffe%CC%81rens+travaux+relatifs+aux+me%CC%81tiers%2C+%26c.+%3A+de%CC%81die%CC%81+au+Roi&rft.oclc=1340470817&rft.iteminfo5=23524275850006421&rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Abook&rft_id=https%3A%2F%2Fbibdata.princeton.edu%2Fbibliographic%2F9947127883506421&rft_id=info%3Aoclcnum%2F1340470817&rfr_id=info%3Asid%2Fcatalog.princeton.edu%3Agenerator&CallNumber=26204+Euro+18Q&ItemInfo1=Reading+Room+Access+Only&ItemNumber=32101071666141&Location=rare%24ctsn&ReferenceNumber=9947127883506421&Site=RBSC
```

### Implementation notes, if any
- There have been some recent changes to the Aeon requests code, including https://github.com/pulibrary/orangelight/pull/3577
- Limit titles to 250 characters
- Limit publishers to 250 characters
- There is a CtxBuilder class that creates the openurl ctx for the Aeon requests and also the CoINS zotero span.  We could truncate the title and publisher in that class, and it would affect both scenarios, which is fine.

**Comment by maxkadel** [comment]
There is [code that references capping the title field length](https://github.com/pulibrary/orangelight/blob/a3abe04850c10d3e99c33f674c926814878e619b/spec/models/requests/requestable_spec.rb#L592), but that does not appear to be happening for this object.

**Comment by sandbergja** [comment]
closed by #3606
